testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/javascript/JavascriptAdapterTest.java

```diff
@@ -822,6 +822,42 @@ public void check3pCookiesMessageCallbackTest() {
                 .init(defaultArguments(), this::assertInitNotAuth);
     }
 
+    // In case of incorrect/unavailable realm provided in KeycloakConfig,
+    // JavaScript Adapter init() should fail-fast and reject Promise with KeycloakError.
+    @Test
+    public void checkInitWithInvalidRealm() {
+
+        JSObjectBuilder keycloakConfig = JSObjectBuilder.create()
+                .add("url", authServerContextRootPage + "/auth")
+                .add("realm", "invalid-realm-name")
+                .add("clientId", CLIENT_ID);
+
+        JSObjectBuilder initOptions = defaultArguments();
+
+        testExecutor
+                .configure(keycloakConfig)
+                .init(initOptions, this::assertErrorResponse);
+
+    }
+
+    // In case of unavailable Authorization Server due to network or other kind of problems,
+    // JavaScript Adapter init() should fail-fast and reject Promise with KeycloakError.
+    @Test
+    public void checkInitWithUnavailableAuthServer() {
+
+        JSObjectBuilder keycloakConfig = JSObjectBuilder.create()
+                .add("url", "https://localhost:12345/auth")
+                .add("realm", REALM_NAME)
+                .add("clientId", CLIENT_ID);
+
+        JSObjectBuilder initOptions = defaultArguments();
+
+        testExecutor
+                .configure(keycloakConfig)
+                .init(initOptions, this::assertErrorResponse);
```

Comment:
```suggestion
                .init(initOptions, assertErrorResponse("Keycloak server is not reachable, most likely due to network problems"));
```

Reply:
I did this intentionally, because it will success only inside this integration test, not in real world, due to different CORS environment. When running JavaScript test suite together with  default PhantomJS browser, CORS is allowed with PhantomJS' `--web-security=false` option. So XHR is not blocked on 404 HTTP error response. The same behavior I observe when running tests with Firefox browser. Not sure why CORS is allowed in this case, maybe because JS test code and local Authorization Server runs under same origin...

Commands I've used to run tests:
```
$ mvn -s maven-settings.xml -f testsuite/integration-arquillian/tests/base/pom.xml clean install -Dtest=JavascriptAdapterTest#checkInitWithInvalidRealm+checkInitWithUnavailableAuthServer -Pauth-server-wildfly

$ mvn -s maven-settings.xml -f testsuite/integration-arquillian/tests/base/pom.xml clean install -Dtest=JavascriptAdapterTest#checkInitWithInvalidRealm+checkInitWithUnavailableAuthServer -Pauth-server-wildfly -Dbrowser=firefox -Djs.browser=firefox -DfirefoxHeadless=true -Dwebdriver.gecko.driver=$HOME/.local/bin/geckodriver
```

So I've decided not to check error details and error message in tests, only fact that KeycloakError is returned from Promise, because this fine-grained details will not be available nowadays in real world scenario where errors are returned by Authorization Server without CORS headers and rejected by XHR.

So as you wish, we can make this changes and check error message contents in tests.

Reply:
I see, ok I didn't know that, thanks. In that case, we can leave it as is.